TezosKit/TezosNode/Services/RPCResponseHandler.swift

```diff
@@ -27,28 +27,18 @@ public class RPCResponseHandler {
 
     // Check for a generic error on the request. If so, propagate.
     if let error = error {
-       let desc = error.localizedDescription
-       let rpcError = TezosKitError(kind: .rpcError, underlyingError: desc)
+      let description = error.localizedDescription
+      let rpcError = TezosKitError.rpcError(description: description)
       return .failure(rpcError)
     }
 
-    // Check for data
-	guard let data = data, let stringData = String(data: data, encoding: .utf8) else {
-	  let tezosKitError = TezosKitError(kind: .unexpectedResponse, underlyingError: nil)
-	  return .failure(tezosKitError)
-	}
-
-	// Check for \"status\":\"backtracked\" in the response, indicating a failed transaction being rolledback
-	if stringData.contains("\"status\":\"backtracked\"") {
```

Comment:
@keefertaylor can you try take the code I had for catching backtracks in the other PR and replace this block with it. Feel free to refactor again, update, rename etc. But I can't update to the latest code until thats in it as it will break our app. Once you have that you can close my error handling PR entirely.

Also I came across another issue that mine wasn't catching either. If you use a customFeePolicy and set it too low (by accident). Because the estimation is never called it doesn't catch gas errors. When that happens you get a `.success(let opHash)` back from the tezosKit `send`, but the console will have "protoxxx.gasExhausted....." in it. Would be great to try catch that too

Reply:
Acked on the gasExhaused. Filed #190